packages/lib-classifier/src/store/ClassificationStore.js

```diff
@@ -158,28 +158,14 @@ const ClassificationStore = types
     }
 
     function * submitClassification (classification) {
-      console.log('Saving classification')
-      const root = getRoot(self)
-      const client = root.client.panoptes
       self.loadingState = asyncStates.posting
 
-      if (!isServiceWorkerAvailable() || !isBackgroundSyncAvailable()) {
-        // Use fallback queuing class
-        self.classificationQueue.add(classification)
-      } else {
-        try {
-          const response = yield client.post(`/${self.type}`, { classifications: classification })
-          if (response.ok) {
-            const savedClassification = response.body.classifications[0]
-            console.log(`Saved classification ${savedClassification.id}`)
-            // TODO: instead of callback here, let's add a Split store that uses an observer
-            self.onClassificationSaved(savedClassification)
-            self.loadingState = asyncStates.success
-          }
-        } catch (error) {
-          console.error(error)
-          self.loadingState = asyncStates.error
-        }
+      // Service worker isn't working right now, so let's use the fallback queue for all browsers
+      try {
+        yield self.classificationQueue.add(classification)
```

Comment:
Is `yield` needed here? Will it stop us from continuing to classify until responses have been received for all the queued classifications?

Reply:
I double checked how this is working. I thought for a moment we could remove it because, yes yield would pause the generator until the promise resolves from adding the classification to the queue and that promise already is doing its own error catching, however, when I removed the async flow wrapper from this action, the other actions for resetting the state and creating a new classification for the next subject didn't function as expected. They would not fire or fire multiple times. I don't entirely understand this, but `flow` seems to do something internally with the MST store actions more than just wrapping an async action with a generator.